selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java

```diff
@@ -428,4 +684,107 @@ public void waitPageLoad(int timeout) {
   public void waitPageLoad() {
     waitPageLoad(DEFAULT_TIMEOUT);
   }
+
+  public List<String> getAvailableStacks() {
+    return seleniumWebDriverHelper
+        .waitPresenceOfAllElements(By.xpath("//div[@data-stack-id]"))
+        .stream()
+        .map(webElement -> webElement.getAttribute("data-stack-id"))
+        .collect(toList());
+  }
+
+  public void waitStackSelected(Stack stack) {
+    String selectedStackXpath =
+        format(
+            "//div[@data-stack-id='%s' and contains(@class, 'stack-selector-item-selected')]",
+            stack.getId());
+    seleniumWebDriverHelper.waitVisibility(By.xpath(selectedStackXpath));
+  }
+
+  public List<String> getVisibleStacks() {
+    return seleniumWebDriverHelper
+        .waitPresenceOfAllElements(By.xpath("//div[@data-stack-id]"))
+        .stream()
+        .filter(seleniumWebDriverHelper::isVisible)
+        .map(webElement -> webElement.getAttribute("data-stack-id"))
+        .collect(Collectors.toList());
+  }
+
+  public void waitVisibleStacks(List<String> expectedVisibleStacks) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> getVisibleStacks().equals(expectedVisibleStacks));
+  }
+
+  public void waitStacks(List<String> expectedStacks) {
+    expectedStacks.forEach(
+        stackId ->
+            seleniumWebDriverHelper.waitPresence(
+                By.xpath(format("//div[@data-stack-id='%s']", stackId))));
+  }
+
+  public int getAvailableStacksCount() {
+    return getAvailableStacks().size();
+  }
+
+  public void waitStacksNotPresent(List<String> stacksIdForChecking) {
+    stacksIdForChecking.forEach(
+        stackId ->
+            webDriverWaitFactory
+                .get()
+                .until(
+                    (ExpectedCondition<Boolean>)
+                        driver -> !getAvailableStacks().contains(stackId)));
+  }
+
+  public void waitStacksOrder(List<String> expectedOrder) {
+    webDriverWaitFactory
+        .get()
+        .until((ExpectedCondition<Boolean>) driver -> expectedOrder.equals(getAvailableStacks()));
+  }
+
+  public void clickNameButton() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath("//div[@che-column-title='Name']/div"));
+  }
+
+  public void clickOnTitlePlaceCoordinate() {
+    seleniumWebDriverHelper.moveCursorTo(
+        seleniumWebDriverHelper.waitPresence(By.id(TOOLBAR_TITLE_ID)));
+    seleniumWebDriverHelper.getAction().click().perform();
+  }
+
+  public void clickOnInputFieldTag(String tagName) {
+    seleniumWebDriverHelper.waitAndClick(
+        By.xpath(format(FILTERS_INPUT_TAG_XPATH_TEMPLATE, tagName)));
+  }
+
+  public void clickOnAddStackButton() {
+    seleniumWebDriverHelper.waitAndClick(By.id("add-stack-button"));
+  }
+
+  public void waitCreateStackDialog() {
+    WaitUtils.sleepQuietly(2);
+    seleniumWebDriverHelper.waitVisibility(By.xpath("//md-dialog"));
```

Comment:
Why do we use sleep?

Reply:
I have added an issue for resolving this problem.